### PR TITLE
Add Insteon onboarding support.

### PR DESCRIPTION
**Pull request by m-chau**

### CODE REVIEW

**Comment by m-chau** [comment]
Passed with CLI.

File: org.opent2t.onboarding.insteon/org.opent2t.onboarding.insteon.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Object Introspection 1.0//EN" "http://standards.freedesktop.org/dbus/introspect-1.0.dtd"[]>
+<!--  This is the metadata for a category of things that are connected via the Insteon Hub. -->
+<node>
+    <interface name="org.opent2t.onboarding.insteon">
```
Comment (avneet723) [inline org.opent2t.onboarding.insteon/org.opent2t.onboarding.insteon.xml]: should be <interface name="org.opent2t.onboarding.insteon**hub**">

File: org.opent2t.onboarding.insteon/js/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "opent2t-onboarding-org-opent2t-onboarding-insteon",
```
Comment (arjun-msft) [inline org.opent2t.onboarding.insteon/js/package.json]: Actually this is the right name.

Comment (arjun-msft) [inline org.opent2t.onboarding.insteon/js/package.json]: Nest and wink are the same way.

The way the opent2t library works, it starts with opent2t and then adds onboarding and then adds what's in the manifest, which is org.opent2t.onboarding.insteon. So you need the double level. 

I think this should be changed to what you suggested in the near future.

**Comment by arjun-msft** [comment]
:shipit:

File: org.opent2t.onboarding.insteon/js/common.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/* jshint esversion: 6 */
+/* jshint node: true */
```
Comment (ChuckFerring) [inline org.opent2t.onboarding.insteon/js/common.js]: Needed?

File: org.opent2t.onboarding.insteon/js/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "version": "1.0.0",
+  "description": "Onboarding for Insteon",
+  "license": "MIT",
+  "main": "thingOnboarding.js",
+  "files": [
+    "thingOnboarding.js",
+    "common.js",
+    "package.json"
```
Comment (ChuckFerring) [inline org.opent2t.onboarding.insteon/js/package.json]: >"package.json" [](start = 4, length = 14)

Is it necessary to include this file?

File: org.opent2t.onboarding.insteon/js/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Insteon Hub Onboarding
+Sample [Insteon Hub](http://www.insteon.com/) onboarding module for Open Translators to Things. May be used to discover devices connected to a Insteon Hub per the 
+[org.opent2t.onboarding.insteon.xml](https://github.com/opent2t/onboarding/blob/master/org.opent2t.onboarding.insteon/org.opent2t.onboarding.insteon.xml) schema.
+
+See schema definition for inputs to the onboarding method.
```
Comment (ChuckFerring) [inline org.opent2t.onboarding.insteon/js/README.md]: You should add instructions for how to run the tests in the readme.

File: org.opent2t.onboarding.insteon/js/thingOnboarding.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/* jshint esversion: 6 */
+/* jshint node: true */
```
Comment (ChuckFerring) [inline org.opent2t.onboarding.insteon/js/thingOnboarding.js]: Needed?

File: org.opent2t.onboarding.insteon/js/tests/test.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+var test = require('ava');
+var OpenT2T = require('opent2t').OpenT2T;
+var testConfig = require('./testConfig');
+var onboardingPath = require('path').join(__dirname, '..');
+
+console.log("Test Config:");
+console.log(JSON.stringify(testConfig, null, 2));
+
+///
+/// Run a series of tests to validate the translator
```
Comment (ChuckFerring) [inline org.opent2t.onboarding.insteon/js/tests/test.js]: >translator [](start = 42, length = 10)

onboarding.

**Comment by ChuckFerring** [comment]
:clock1:

File: org.opent2t.onboarding.insteon/js/common.js (hunk 3 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/common.js]: I don't think we need it. But all other hubs has it comment out and this goes back to Arjun's comments on we should probably keep everything consistent now and remove it later all together.

File: org.opent2t.onboarding.insteon/js/package.json (hunk 4 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/package.json]: I am not sure but other hubs included these files. @arjun-msft  for comments.

File: org.opent2t.onboarding.insteon/js/thingOnboarding.js (hunk 6 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/thingOnboarding.js]: don't think we need it. But all other hubs has it comment out and this goes back to Arjun's comments on we should probably keep everything consistent now and remove it later all together.

File: org.opent2t.onboarding.insteon/js/README.md (hunk 5 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/README.md]: Added.

File: org.opent2t.onboarding.insteon/js/common.js (hunk 3 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/common.js]: Removed.

---
In reply to: [82906938](https://github.com/openT2T/onboarding/pull/21#discussion_r82906938) [](ancestors = 82906938)

File: org.opent2t.onboarding.insteon/js/thingOnboarding.js (hunk 6 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/thingOnboarding.js]: Removed.

---
In reply to: [82907678](https://github.com/openT2T/onboarding/pull/21#discussion_r82907678) [](ancestors = 82907678)

File: org.opent2t.onboarding.insteon/js/tests/test.js (hunk 7 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/tests/test.js]: Fixed.

---
In reply to: [82907798](https://github.com/openT2T/onboarding/pull/21#discussion_r82907798) [](ancestors = 82907798)

File: org.opent2t.onboarding.insteon/js/README.md (hunk 5 above)
Comment (m-chau) [inline org.opent2t.onboarding.insteon/js/README.md]: Added.

---
In reply to: [82907511](https://github.com/openT2T/onboarding/pull/21#discussion_r82907511) [](ancestors = 82907511)

**Pull request merged**